application/src/main/java/org/togetherjava/tjbot/commands/moderation/scam/ScamBlocker.java

```diff
@@ -0,0 +1,364 @@
+                ModerationAction.QUARANTINE, null, reason);
+
+        guild
+            .addRoleToMember(author,
+                    ModerationUtils.getQuarantinedRole(guild, config).orElseThrow())
+            .reason(reason)
+            .queue();
+    }
+
+    private void reportScamMessage(@NotNull GuildMessageReceivedEvent event,
+            @NotNull String reportTitle, @NotNull Collection<? extends ActionRow> actionRows) {
+        Guild guild = event.getGuild();
+        Optional<TextChannel> reportChannel = getReportChannel(guild);
+        if (reportChannel.isEmpty()) {
+            logger.warn(
+                    "Unable to report a scam message, did not find a report channel matching the configured pattern '{}' for guild '{}'",
+                    reportChannelPattern, guild.getName());
+            return;
+        }
+
+        User author = event.getAuthor();
+        MessageEmbed embed =
+                new EmbedBuilder().setDescription(event.getMessage().getContentStripped())
+                    .setTitle(reportTitle)
+                    .setAuthor(author.getAsTag(), null, author.getAvatarUrl())
+                    .setTimestamp(event.getMessage().getTimeCreated())
+                    .setColor(AMBIENT_COLOR)
+                    .setFooter(author.getId())
+                    .build();
+        Message message = new MessageBuilder().setEmbeds(embed).setActionRows(actionRows).build();
+
+        reportChannel.orElseThrow().sendMessage(message).queue();
+    }
+
+    private void dmUser(@NotNull GuildMessageReceivedEvent event) {
+        dmUser(event.getGuild(), event.getAuthor().getIdLong(), event.getJDA());
+    }
+
+    private void dmUser(@NotNull Guild guild, long userId, @NotNull JDA jda) {
+        String dmMessage =
+                """
+                        Hey there, we detected that you did send scam in the server %s and therefore put you under quarantine.
+                        This means you can no longer interact with anyone in the server until you have been unquarantined again.
+
+                        If you think this was a mistake (for example, your account was hacked, but you got back control over it),
+                        please contact a moderator or admin of the server.
+                        """
+                    .formatted(guild.getName());
+
+        jda.openPrivateChannelById(userId)
+            .flatMap(channel -> channel.sendMessage(dmMessage))
+            .queue();
+    }
+
+    private Optional<TextChannel> getReportChannel(@NotNull Guild guild) {
+        // Check cache first, then get full list
+        return guild.getTextChannelCache()
+            .stream()
+            .filter(isReportChannel)
+            .findAny()
+            .or(() -> guild.getTextChannels().stream().filter(isReportChannel).findAny());
+    }
+
+    private List<ActionRow> createConfirmDialog(@NotNull GuildMessageReceivedEvent event) {
```

Comment:
Use addActionRow instead of addActionRows

Reply:
@Tais993 That doesnt exist for the situation im in here.

Reply:
Still don't think a method should return a List, when it only has 1 element.

Reply:
Regardless, there's always a non-collection overload.

Reply:
Checked the code. The reason why its a collection is bc the result is handled around a few times before it is actually used.

Methods like `takeActionAutoDeleteButApproveQuarantine` call `createConfirmDialog` and pass the list to `reportScamMessage`, which then does `new MessageBuilder()....setActionRows(actionRows)...`. But the method can also be called with an empty list if no dialog is desired.

I made the List a Collection now though.

The only other solution I see is to change it to varargs mixed with arrays. Imo thats uglier, in particular since I need an empty array for no-dialog then. Not so nice to create.

Reply:
I still want createConfirmDialog to return an ActionRow, the methods that call it can make it a list.
createConfirmDialog doesn't need a List, it's useless

Reply:
But what if a confirm dialog might consist of multiple rows? Kinda an implementation detail, isnt it? Why should the method enforce that its only a single row.

I dont have a strong preference here, I can make it a single `@Nullable ActionRow` as well. But then I probably have to do an if-else to either call the `addActionRow` or not, depending on whether its null (unless JDA handles null gracefully here, have to check the code).

Reply:
> But what if a confirm dialog might consist of multiple rows? Kinda an implementation detail, isnt it? Why should the method enforce that its only a single row.

If that happens in the future, I guess we have to change our code. 
Since the dialog is only 1 row, it's easy for the calling methods to add their own rows too, otherwise they need to do weird checks, or move rows. So regarding future compatibility, I'd argue that 1 actionrow is better.

Now the dialog just creates the row, and that is all that it needs to do.

>  I can make it a single @Nullable ActionRow as wel

Why would it be nullable randomly? It's never null? 
And if it's null, JDA would throw too if you give it a Collection with a null entry.

Reply:
done